Fixes issue 1026

JoshuaMcManus opened the pull request:
The equals() and hash() functions for keys within a class now exclusively call get methods
This prevents the issue of trying to access private inherited members directly while also keeping a consistent format

https://github.com/umple/umple/issues/1026

**IMPORTANT**: By submitting a patch, you agree that your work will be licensed under the [license](https://github.com/umple/umple/blob/master/LICENSE.md) used by the project.

If your pull request does not pass CI, it will **not** be merged (unless *heavily* justified). 

## Description

Applies a fix to issue 1026

## Tests

Had to change existing tests since they were expecting java code to be produced which directly accesses member variables (we now only call the gettters). All other tests passed and those that were modified now pass as well.

Closes #1026  
Closes #1089

Commit:
Fixes issue 1026

The equals() and hash() functions for keys within a class now exclusively call get methods
This prevents the issue of trying to access private inherited members directly while also keeping a consistent format

https://github.com/umple/umple/issues/1026

TimLethbridge reviewed (changes requested):
THis PR needs one or two additional tests for the code that resulted in the issue to start with. In other words there should be a test for code
```
class Test{
  id;
}

class Sub{
  isA Test;
  key {id}
}
```
demonstrating that it parses and/or generates correct output.

Commit:
fixes-1089 and adds tests

Adds some testbed tests for the key attribute to check issues fixed in this PR
Fixes an issue where an array of Integers was not properly being compared between objects

Changes a test which expects a null String to be different than an empty String to now expect it to be equal within an object. This is because the existing get() method for String attributes returns an empty string for null values and we are now using get() methods instead of accessing the variables directly

Commit:
Adds more tests and cleans up existing

Adds another test and extends an existing one to verify the java outputted in the equals function is syntactically correct
Cleans up the naming of previously committed tests

Commit:
adds role name to association in test
